specs/_features/focil/beacon-chain.md

```diff
@@ -0,0 +1,278 @@
+
+*Note:* This specification is built upon [Electra](../../electra/beacon_chain.md) and is under active development.
+
+## Constants
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_IL_COMMITTEE`       | `DomainType('0x0C000000')`  # (New in FOCIL)|
+
+## Preset
+
+### Inclusion List Committee
+
+| Name | Value | 
+| - | - | 
+| `IL_COMMITTEE_SIZE` | `uint64(2**9)` (=256)  # (New in FOCIL) |
+
+### Execution
+
+| Name | Value |
+| - | - |
+| `MAX_TRANSACTIONS_PER_INCLUSION_LIST` |  `uint64(1)` # (New in FOCIL) TODO: Placeholder | 
+
+## Containers
+
+### New containers
+
+#### `InclusionSummary`
+
+```python
+class InclusionSummary(Container):
+    address: ExecutionAddress
+    nonce: uint64
+    gas_limit: uint64
+```
+
+#### `LocalInclusionList`
+
+```python
+class LocalInclusionList(Container):
+    slot: Slot
+    validator_index: ValidatorIndex
+    parent_hash: Hash32
+    summaries: List[InclusionSummary, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+    transactions: List[Transaction, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
+```
+
+#### `SignedLocalInclusionList`
+
+```python
+class SignedLocalInclusionList(Container):
+    message: LocalInclusionList
+    signature: BLSSignature
+```
+
+#### `InclusionSummaryAggregate`
+
+```python
+class InclusionSummaryAggregate(Container):
+    aggregation_bits: Bitvector[IL_COMMITTEE_SIZE]
+    summary: InclusionSummary
+    signature: BLSSignature
+```
+
+#### `IndexedInclusionSummaryAggregate`
+
+```python
+class IndexedInclusionSummaryAggregate(Container):
+    validator_indices: List[ValidatorIndex, IL_COMMITTEE_SIZE]
+    summary: InclusionSummary
+    signature: BLSSignature
+```
+
+#### `InclusionSummaryAggregates`
+
+```python
+class InclusionSummaryAggregates(Container):
+    List[InclusionSummaryAggregate, MAX_TRANSACTIONS_PER_INCLUSION_LIST * IL_COMMITTEE_SIZE]
+```
+
+### Modified containers
+
+#### `BeaconBlockBody`
+
+**Note:** The Beacon Block body is modified to contain a new `inclusion_summary_aggregates` field.
+
+```python
+class BeaconBlockBody(Container):
+    randao_reveal: BLSSignature
+    eth1_data: Eth1Data  # Eth1 data vote
+    graffiti: Bytes32  # Arbitrary data
+    # Operations
+    proposer_slashings: List[ProposerSlashing, MAX_PROPOSER_SLASHINGS]
+    attester_slashings: List[AttesterSlashing, MAX_ATTESTER_SLASHINGS_ELECTRA]
+    attestations: List[Attestation, MAX_ATTESTATIONS_ELECTRA]
+    deposits: List[Deposit, MAX_DEPOSITS]
+    voluntary_exits: List[SignedVoluntaryExit, MAX_VOLUNTARY_EXITS]
+    sync_aggregate: SyncAggregate
+    bls_to_execution_changes: List[SignedBLSToExecutionChange, MAX_BLS_TO_EXECUTION_CHANGES]
+    # FOCIL
+    inclusion_summary_aggregates: InclusionSummaryAggregates   # [New in FOCIL]
```

Comment:
I just realize this could add quite amount of bytes to beacon block. I'm not aware of what is the acceptable range, but here is my back-of-the-napkin math.

InclusionSummary: address + uint64 + uint64 = (20 + 8 + 8) bytes = 36 bytes.

InclusionSummaryAggregate: Bitvector[`IL_COMMITTEE_SIZE`] + InclusionSummary + BLSSignature = (256 / 8 + 36 + 96) bytes = 164 bytes.

InclusionSummaryAggregates: InclusionSummaryAggregate * `MAX_TRANSACTIONS_PER_INCLUSION_LIST` * `IL_COMMITTEE_SIZE` = (164 * `MAX_TRANSACTIONS_PER_INCLUSION_LIST` * 256) bytes >= 41,984 bytes.

Of course it depends on constants values, which are not fixed yet.

The ideal size of the new field would be (128 + 36 * `MAX_TRANSACTIONS_PER_INCLUSION_LIST`) bytes, which comes from: 
```
aggregation_bits: BitVector[IL_COMMITTEE_SIZE]
summaries: List[InclusionSummary, MAX_TRANSACTIONS_PER_INCLUSION_LIST]
signature: BLSSignature
```

But this requires at least one more round from a proposer to IL committee members. I don't have a good solution at the moment but wanted to share what I've found.